### PR TITLE
NMS-12962: Some restructuring and bug fixes

### PR DESCRIPTION
**Pull request by cgorantla**
This PR involves
1. Update indexes for tables.
2.  Move Adapter to use AbstractAdapter to remove the dependency of collection configs on Sentinel
3.  Enhancements to routeinfo/asninfo scheduling.

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-12962

### CODE REVIEW

**Review by RangerRick — APPROVED** [review]
I can't say I have much of an understanding of BMP or BGP yet, but the code looks reasonable to me +1

File: features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/BmpAdapterCommon.java [hunk 1]
```diff
@@ -0,0 +1,808 @@
...
+        router.sequence = sequence.getAndIncrement();
+        router.name = null;
+        router.hash = context.routerHashId;
+        router.ipAddress = context.sourceAddress;
+        router.description = null;
+        router.termCode = termination.getReason();
+
+        switch (router.termCode) {
+            case 0:
+                router.termReason = "Session administratively closed.  The session might be re-initiated";
```
Comment (christianpape) [inline features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/BmpAdapterCommon.java:304]: Two spaces `...closed.  The...`

File: features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/BmpAdapterCommon.java [hunk 2]
```diff
@@ -0,0 +1,808 @@
...
+                break;
+            case 1:
+                router.termReason = "Unspecified reason";
+                break;
+            case 2:
+                router.termReason = "Out of resources.  The router has exhausted resources available for the BMP session";
```
Comment (christianpape) [inline features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/BmpAdapterCommon.java:310]: same here.

File: features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/BmpAdapterCommon.java [hunk 3]
```diff
@@ -0,0 +1,808 @@
...
+                break;
+            case 3:
+                router.termReason = "Redundant connection.  The router has determined that this connection is redundant with another one";
```
Comment (christianpape) [inline features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/BmpAdapterCommon.java:313]: and here.

**Review by christianpape — APPROVED** [review]
As far as I can see at the moment, things are looking good.

**Comment by cgorantla** [comment]
Thanks @RangerRick @christianpape  for the review. Since last commit is only LOG update, I'm going ahead and merging this.

**Pull request merged**